// sabik-datagrid/dist/index.js
'use strict';var Oe=require('react'),reactTable=require('@tanstack/react-table'),jsxRuntime=require('react/jsx-runtime'),reactVirtual=require('@tanstack/react-virtual');function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var Oe__default=/*#__PURE__*/_interopDefault(Oe);var te=`/* sabik-datagrid \u2014 self-contained styles (no Tailwind required) */
.sdg {
  --sdg-bg: #ffffff;
  --sdg-fg: #0f172a;
  --sdg-muted: #f1f5f9;
  --sdg-muted-fg: #64748b;
  --sdg-border: #e2e8f0;
  --sdg-input: #e2e8f0;
  --sdg-primary: #2563eb;
  --sdg-primary-fg: #ffffff;
  --sdg-destructive: #dc2626;
  --sdg-destructive-bg: #fef2f2;
  --sdg-row-hover: #f8fafc;
  --sdg-row-selected: rgba(37, 99, 235, 0.1);
  --sdg-shadow: 0 1px 2px rgba(15, 23, 42, 0.06);
  --sdg-radius: 12px;
  --sdg-font: system-ui, -apple-system, Segoe UI, Roboto, sans-serif;

  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  font-family: var(--sdg-font);
  color: var(--sdg-fg);
  background: var(--sdg-bg);
  border: 1px solid var(--sdg-border);
  border-radius: var(--sdg-radius);
  box-shadow: var(--sdg-shadow);
  overflow: hidden;
}

.sdg *,
.sdg *::before,
.sdg *::after {
  box-sizing: border-box;
}

.sdg--fullscreen {
  position: fixed;
  inset: 0;
  z-index: 50;
  padding: 1.5rem;
  border-radius: 0;
}

.sdg__chrome {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: var(--sdg-bg);
}

.sdg__icon-btn {
  padding: 0.75rem;
  margin-right: 0.5rem;
  color: var(--sdg-muted-fg);
  background: transparent;
  border: none;
  border-radius: 0.5rem;
  cursor: pointer;
}

.sdg__icon-btn:hover {
  color: var(--sdg-fg);
  background: var(--sdg-muted);
}

.sdg__scroll {
  position: relative;
  overflow: auto;
  flex: 1;
  min-height: 0;
}

/* flex:1 + height:100% parent collapses the viewport; virtual mode needs a fixed scrollport */
.sdg__scroll--virtual {
  flex: none;
  flex-grow: 0;
  flex-shrink: 0;
  height: 400px;
  min-height: 400px;
  max-height: 400px;
  overflow: auto;
}

.sdg__scroll--sticky-header {
  max-height: 100%;
}

.sdg__table {
  width: 100%;
  text-align: left;
  border-collapse: collapse;
}

.sdg__thead {
  background: color-mix(in srgb, var(--sdg-muted) 80%, transparent);
  color: var(--sdg-muted-fg);
  font-weight: 500;
  font-size: 0.75rem;
  text-transform: uppercase;
  letter-spacing: 0.04em;
}

.sdg__th {
  border-bottom: 1px solid var(--sdg-border);
  vertical-align: middle;
  position: relative;
}

.sdg__th-inner {
  display: flex;
  align-items: center;
  justify-content: space-between;
  cursor: default;
  user-select: none;
}

.sdg__th-inner--sortable {
  cursor: pointer;
}

.sdg__th-inner--sortable:hover {
  color: var(--sdg-fg);
}

.sdg__th-label {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.sdg__sort {
  margin-left: 0.5rem;
  display: flex;
  flex-direction: column;
  font-size: 8px;
  opacity: 0.5;
  line-height: 1;
}

.sdg__sort-active {
  color: var(--sdg-primary);
  font-weight: 700;
  opacity: 1;
}

.sdg__resize {
  position: absolute;
  right: 0;
  top: 0;
  height: 100%;
  width: 4px;
  cursor: col-resize;
}

.sdg__tbody {
  border-top: 1px solid var(--sdg-border);
}

.sdg__tr {
  transition: background-color 0.15s ease;
  cursor: pointer;
}

.sdg__tr:hover {
  background: var(--sdg-row-hover);
}

.sdg__tr--selected {
  background: var(--sdg-row-selected);
}

.sdg__tr--zebra:nth-child(even) {
  background: #f8fafc;
}

.sdg__td {
  border-right: 1px solid color-mix(in srgb, var(--sdg-border) 50%, transparent);
  font-size: 0.875rem;
  color: color-mix(in srgb, var(--sdg-fg) 80%, transparent);
}

.sdg__td:last-child {
  border-right: none;
}

.sdg__td--actions {
  border-left: 1px solid color-mix(in srgb, var(--sdg-border) 50%, transparent);
  text-align: center;
}

.sdg__cell--compact {
  padding: 0.25rem 0.5rem;
  font-size: 0.75rem;
}

.sdg__cell--comfortable {
  padding: 0.5rem 1rem;
  font-size: 0.875rem;
}

.sdg__cell--spacious {
  padding: 1rem 1.5rem;
  font-size: 1rem;
}

.sdg__checkbox {
  border-radius: 0.25rem;
  accent-color: var(--sdg-primary);
}

.sdg__toolbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.75rem 1rem;
  background: var(--sdg-bg);
  border-bottom: 1px solid var(--sdg-border);
  gap: 1rem;
  position: relative;
  flex: 1;
}

.sdg__toolbar-group {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.sdg__search {
  position: relative;
}

.sdg__search-icon {
  position: absolute;
  left: 0.75rem;
  top: 50%;
  transform: translateY(-50%);
  color: var(--sdg-muted-fg);
  display: flex;
  pointer-events: none;
}

.sdg__search-input {
  padding: 0.5rem 1rem 0.5rem 2.25rem;
  font-size: 0.875rem;
  border: 1px solid var(--sdg-input);
  border-radius: 0.5rem;
  width: 16rem;
  max-width: 100%;
  background: var(--sdg-bg);
  color: var(--sdg-fg);
  outline: none;
}

.sdg__search-input:focus {
  border-color: var(--sdg-primary);
  box-shadow: 0 0 0 3px rgba(37, 99, 235, 0.15);
}

.sdg__btn {
  padding: 0.5rem 0.75rem;
  font-size: 0.75rem;
  font-weight: 500;
  border: 1px solid var(--sdg-input);
  border-radius: 0.5rem;
  background: var(--sdg-bg);
  color: var(--sdg-fg);
  cursor: pointer;
}

.sdg__btn:hover {
  background: var(--sdg-muted);
}

.sdg__btn:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

.sdg__btn--primary {
  background: var(--sdg-primary);
  color: var(--sdg-primary-fg);
  border-color: var(--sdg-primary);
}

.sdg__btn--primary:hover {
  opacity: 0.9;
  background: var(--sdg-primary);
}

.sdg__btn--danger {
  background: var(--sdg-destructive-bg);
  color: var(--sdg-destructive);
  border-color: color-mix(in srgb, var(--sdg-destructive) 25%, transparent);
}

.sdg__btn--ghost {
  border: none;
  color: var(--sdg-muted-fg);
}

.sdg__btn--ghost:hover {
  color: var(--sdg-fg);
  background: transparent;
}

.sdg__pagination {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.75rem 1rem;
  background: var(--sdg-bg);
  border-top: 1px solid var(--sdg-border);
  font-size: 0.875rem;
  color: var(--sdg-muted-fg);
  font-weight: 500;
}

.sdg__pagination-meta {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.sdg__pagination-meta strong,
.sdg__pagination-meta .sdg__page-num {
  color: var(--sdg-fg);
  font-weight: 600;
}

.sdg__pagination-controls {
  display: flex;
  align-items: center;
  gap: 0.75rem;
}

.sdg__select {
  background: var(--sdg-bg);
  border: 1px solid var(--sdg-input);
  border-radius: 0.375rem;
  padding: 0.25rem 0.5rem;
  font-size: 0.75rem;
  color: var(--sdg-fg);
  outline: none;
}

.sdg__bulk {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.5rem 1rem;
  background: rgba(37, 99, 235, 0.05);
  border-bottom: 1px solid rgba(37, 99, 235, 0.2);
}

.sdg__bulk-meta {
  display: flex;
  align-items: center;
  gap: 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
  color: var(--sdg-primary);
}

.sdg__bulk-count {
  background: var(--sdg-primary);
  color: var(--sdg-primary-fg);
  padding: 0.125rem 0.5rem;
  border-radius: 999px;
  font-size: 0.75rem;
  font-weight: 700;
}

.sdg__menu {
  position: absolute;
  right: 1rem;
  top: 3rem;
  z-index: 20;
  background: var(--sdg-bg);
  border: 1px solid var(--sdg-border);
  border-radius: 0.375rem;
  box-shadow: 0 8px 24px rgba(15, 23, 42, 0.12);
  padding: 0.5rem;
  width: 12rem;
}

.sdg__menu-title {
  font-size: 0.75rem;
  font-weight: 700;
  margin: 0 0 0.5rem;
  padding: 0 0.5rem;
  color: var(--sdg-muted-fg);
  text-transform: uppercase;
}

.sdg__menu-list {
  display: flex;
  flex-direction: column;
  gap: 0.25rem;
  max-height: 15rem;
  overflow: auto;
}

.sdg__menu-item {
  display: flex;
  align-items: center;
  gap: 0.5rem;
  padding: 0.25rem 0.5rem;
  cursor: pointer;
  font-size: 0.875rem;
  color: var(--sdg-fg);
  border-radius: 0.25rem;
}

.sdg__menu-item:hover {
  background: var(--sdg-muted);
}

.sdg__actions {
  position: relative;
}

.sdg__actions-trigger {
  padding: 0.25rem;
  border: none;
  background: transparent;
  border-radius: 0.25rem;
  cursor: pointer;
  color: var(--sdg-fg);
}

.sdg__actions-trigger:hover {
  background: var(--sdg-muted);
}

.sdg__actions-backdrop {
  position: fixed;
  inset: 0;
  z-index: 10;
}

.sdg__actions-menu {
  position: absolute;
  right: 0;
  margin-top: 0.25rem;
  width: 12rem;
  background: var(--sdg-bg);
  border: 1px solid var(--sdg-border);
  border-radius: 0.375rem;
  box-shadow: 0 8px 24px rgba(15, 23, 42, 0.12);
  z-index: 20;
  padding: 0.25rem 0;
}

.sdg__actions-item {
  width: 100%;
  text-align: left;
  padding: 0.5rem 1rem;
  font-size: 0.875rem;
  border: none;
  background: transparent;
  cursor: pointer;
  color: var(--sdg-fg);
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.sdg__actions-item:hover {
  background: var(--sdg-muted);
}

.sdg__actions-item--danger {
  color: var(--sdg-destructive);
}

.sdg__actions-item--danger:hover {
  background: var(--sdg-destructive-bg);
}

.sdg__actions-item:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.sdg__status {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 16rem;
  width: 100%;
}

.sdg__status-text {
  text-align: center;
  color: var(--sdg-muted-fg);
}

.sdg__status-error {
  text-align: center;
  color: var(--sdg-destructive);
}

.sdg__spinner {
  width: 2rem;
  height: 2rem;
  border: 4px solid var(--sdg-primary);
  border-top-color: transparent;
  border-radius: 50%;
  animation: sdg-spin 0.8s linear infinite;
}

.sdg__skeleton {
  width: 100%;
  animation: sdg-pulse 1.5s ease-in-out infinite;
}

.sdg__skeleton-row {
  display: flex;
  width: 100%;
  border-bottom: 1px solid var(--sdg-border);
}

.sdg__skeleton-row--compact {
  height: 2rem;
}

.sdg__skeleton-row--comfortable {
  height: 3rem;
}

.sdg__skeleton-row--spacious {
  height: 4rem;
}

.sdg__skeleton-cell {
  flex: 1;
  background: #e2e8f0;
  border-radius: 2px;
  margin: 0.5rem;
}

@keyframes sdg-spin {
  to {
    transform: rotate(360deg);
  }
}

@keyframes sdg-pulse {
  0%,
  100% {
    opacity: 1;
  }
  50% {
    opacity: 0.5;
  }
}

@media (prefers-color-scheme: dark) {
  .sdg {
    --sdg-bg: #0f172a;
    --sdg-fg: #f8fafc;
    --sdg-muted: #1e293b;
    --sdg-muted-fg: #94a3b8;
    --sdg-border: #334155;
    --sdg-input: #334155;
    --sdg-row-hover: #1e293b;
    --sdg-row-selected: rgba(37, 99, 235, 0.25);
  }

  .sdg__tr--zebra:nth-child(even) {
    background: #1e293b;
  }

  .sdg__skeleton-cell {
    background: #334155;
  }
}
`;var oe="sabik-datagrid-styles";function Ie(){if(typeof document>"u"||document.getElementById(oe))return;let e=document.createElement("style");e.id=oe,e.textContent=te,document.head.appendChild(e);}Ie();var x={STORAGE_KEY_PREFIX:"sabik-datagrid-persistence:",DENSITY_CLASSES:{compact:"sdg__cell--compact",comfortable:"sdg__cell--comfortable",spacious:"sdg__cell--spacious"}};var F=(e,t,a)=>{let s=e?`${x.STORAGE_KEY_PREFIX}${e}:${t}`:null;return {load:()=>{if(!s)return a;try{let n=localStorage.getItem(s);return n?JSON.parse(n):a}catch(n){return console.error(`Error loading persistence for ${t}:`,n),a}},save:n=>{if(s)try{localStorage.setItem(s,JSON.stringify(n));}catch(l){console.error(`Error saving persistence for ${t}:`,l);}}}};var re=Oe.createContext(null),se=({children:e,persistenceKey:t,initialState:a})=>{let s=F(t,"sorting",[]),o=F(t,"filters",[]),i=F(t,"visibility",{}),n=F(t,"pagination",{pageIndex:0,pageSize:10}),[l,G]=Oe.useState(()=>t?s.load():a?.sorting||[]),[b,p]=Oe.useState(()=>t?o.load():a?.columnFilters||[]),[R,N]=Oe.useState(a?.globalFilter||""),[f,v]=Oe.useState(()=>t?i.load():a?.columnVisibility||{}),[y,u]=Oe.useState(a?.rowSelection||{}),[_,A]=Oe.useState(()=>t?n.load():a?.pagination||{pageIndex:0,pageSize:10}),[z,B]=Oe.useState("comfortable");Oe.useEffect(()=>{t&&(s.save(l),o.save(b),i.save(f),n.save(_));},[l,b,f,_,t]);let I=Oe.useMemo(()=>({state:{sorting:l,columnFilters:b,globalFilter:R,columnVisibility:f,rowSelection:y,pagination:_},setSorting:c=>G(c),setColumnFilters:c=>p(c),setGlobalFilter:c=>N(c),setColumnVisibility:c=>v(c),setRowSelection:c=>u(c),setPagination:c=>A(c),density:z,setDensity:B}),[l,b,R,f,y,_,z]);return jsxRuntime.jsx(re.Provider,{value:I,children:e})},w=()=>{let e=Oe.useContext(re);if(!e)throw new Error("useDataGridContext must be used within a DataGridProvider");return e};var ae=({table:e})=>{let{density:t}=w(),a=x.DENSITY_CLASSES[t];return jsxRuntime.jsx("thead",{className:"sdg__thead",children:e.getHeaderGroups().map(s=>jsxRuntime.jsx("tr",{children:s.headers.map(o=>jsxRuntime.jsxs("th",{colSpan:o.colSpan,className:`sdg__th ${a}`,style:{width:o.getSize()},children:[jsxRuntime.jsxs("div",{className:`sdg__th-inner ${o.column.getCanSort()?"sdg__th-inner--sortable":""}`,onClick:o.column.getToggleSortingHandler(),children:[jsxRuntime.jsx("span",{className:"sdg__th-label",children:reactTable.flexRender(o.column.columnDef.header,o.getContext())}),o.column.getCanSort()&&jsxRuntime.jsxs("div",{className:"sdg__sort",children:[jsxRuntime.jsx("span",{className:o.column.getIsSorted()==="asc"?"sdg__sort-active":"",children:"\u25B2"}),jsxRuntime.jsx("span",{className:o.column.getIsSorted()==="desc"?"sdg__sort-active":"",children:"\u25BC"})]})]}),jsxRuntime.jsx("div",{onMouseDown:o.getResizeHandler(),onTouchStart:o.getResizeHandler(),className:"sdg__resize"})]},o.id))},s.id))})};var V=({row:e,actions:t})=>{let[a,s]=Oe__default.default.useState(false);return jsxRuntime.jsxs("div",{className:"sdg__actions",children:[jsxRuntime.jsx("button",{type:"button",onClick:o=>{o.stopPropagation(),s(!a);},className:"sdg__actions-trigger","aria-haspopup":"menu","aria-expanded":a,children:"\u22EE"}),a&&jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsx("div",{className:"sdg__actions-backdrop",onClick:o=>{o.stopPropagation(),s(false);}}),jsxRuntime.jsx("div",{className:"sdg__actions-menu",role:"menu",children:t.map((o,i)=>jsxRuntime.jsxs("button",{type:"button",role:"menuitem",disabled:o.disabled,onClick:n=>{n.stopPropagation(),o.onClick(e.original),s(false);},className:`sdg__actions-item ${o.danger?"sdg__actions-item--danger":""}`,children:[o.icon,o.label]},i))})]})]})};var ne=({table:e,onRowClick:t,onRowDoubleClick:a,rowActions:s})=>{let{density:o}=w(),i=x.DENSITY_CLASSES[o];return jsxRuntime.jsx("tbody",{className:"sdg__tbody",children:e.getRowModel().rows.map(n=>jsxRuntime.jsxs("tr",{onClick:()=>t?.(n.original),onDoubleClick:()=>a?.(n.original),className:`sdg__tr ${n.getIsSelected()?"sdg__tr--selected":""}`,role:"row","aria-selected":n.getIsSelected(),children:[n.getVisibleCells().map(l=>jsxRuntime.jsx("td",{className:`sdg__td ${i}`,role:"cell",children:reactTable.flexRender(l.column.columnDef.cell,l.getContext())},l.id)),s&&s.length>0&&jsxRuntime.jsx("td",{className:`sdg__td sdg__td--actions ${i}`,children:jsxRuntime.jsx(V,{row:n,actions:s})})]},n.id))})};var de=({table:e,scrollElement:t,onRowClick:a,onRowDoubleClick:s,rowActions:o})=>{let{density:i}=w(),n=x.DENSITY_CLASSES[i],{rows:l}=e.getRowModel(),G=i==="compact"?32:i==="spacious"?64:48,b=reactVirtual.useVirtualizer({count:l.length,getScrollElement:()=>t,estimateSize:()=>G,overscan:8,enabled:!!t}),p=b.getVirtualItems(),R=b.getTotalSize(),N=p.length>0?p[0].start:0,f=p.length>0?R-p[p.length-1].end:0,v=e.getVisibleLeafColumns().length+(o?.length?1:0);return jsxRuntime.jsxs("tbody",{className:"sdg__tbody",children:[N>0&&jsxRuntime.jsx("tr",{"aria-hidden":"true",children:jsxRuntime.jsx("td",{colSpan:v,style:{height:N,padding:0,border:"none"}})}),p.map(y=>{let u=l[y.index];return jsxRuntime.jsxs("tr",{"data-index":y.index,ref:b.measureElement,onClick:()=>a?.(u.original),onDoubleClick:()=>s?.(u.original),className:`sdg__tr sdg__tr--zebra ${u.getIsSelected()?"sdg__tr--selected":""}`,role:"row","aria-selected":u.getIsSelected(),style:{height:`${y.size}px`},children:[u.getVisibleCells().map(_=>jsxRuntime.jsx("td",{className:`sdg__td ${n}`,style:{width:_.column.getSize()},role:"cell",children:reactTable.flexRender(_.column.columnDef.cell,_.getContext())},_.id)),o&&o.length>0&&jsxRuntime.jsx("td",{className:`sdg__td sdg__td--actions ${n}`,children:jsxRuntime.jsx(V,{row:u,actions:o})})]},u.id)}),f>0&&jsxRuntime.jsx("tr",{"aria-hidden":"true",children:jsxRuntime.jsx("td",{colSpan:v,style:{height:f,padding:0,border:"none"}})})]})};var le=({table:e})=>jsxRuntime.jsxs("div",{className:"sdg__pagination",children:[jsxRuntime.jsxs("div",{className:"sdg__pagination-meta",children:[jsxRuntime.jsx("span",{children:"Page"}),jsxRuntime.jsx("span",{className:"sdg__page-num",children:e.getState().pagination.pageIndex+1}),jsxRuntime.jsx("span",{children:"of"}),jsxRuntime.jsx("span",{className:"sdg__page-num",children:e.getPageCount()})]}),jsxRuntime.jsxs("div",{className:"sdg__pagination-controls",children:[jsxRuntime.jsx("select",{value:e.getState().pagination.pageSize,onChange:t=>e.setPageSize(Number(t.target.value)),className:"sdg__select",children:[10,20,50,100].map(t=>jsxRuntime.jsxs("option",{value:t,children:[t," rows / page"]},t))}),jsxRuntime.jsxs("div",{className:"sdg__toolbar-group",children:[jsxRuntime.jsx("button",{type:"button",onClick:()=>e.previousPage(),disabled:!e.getCanPreviousPage(),className:"sdg__btn",children:"Previous"}),jsxRuntime.jsx("button",{type:"button",onClick:()=>e.nextPage(),disabled:!e.getCanNextPage(),className:"sdg__btn",children:"Next"})]})]})]});var ce=({table:e})=>jsxRuntime.jsxs("div",{className:"sdg__menu",children:[jsxRuntime.jsx("p",{className:"sdg__menu-title",children:"Toggle Columns"}),jsxRuntime.jsx("div",{className:"sdg__menu-list",children:e.getAllLeafColumns().map(t=>jsxRuntime.jsxs("label",{className:"sdg__menu-item",children:[jsxRuntime.jsx("input",{type:"checkbox",className:"sdg__checkbox",checked:t.getIsVisible(),onChange:t.getToggleVisibilityHandler()}),String(t.columnDef.header??t.id)]},t.id))})]});var me=({table:e,searchable:t,filterable:a,exportable:s,onExport:o})=>{let[i,n]=Oe__default.default.useState(false);return jsxRuntime.jsxs("div",{className:"sdg__toolbar",children:[jsxRuntime.jsx("div",{className:"sdg__toolbar-group",children:t&&jsxRuntime.jsxs("div",{className:"sdg__search",children:[jsxRuntime.jsx("div",{className:"sdg__search-icon",children:jsxRuntime.jsxs("svg",{xmlns:"http://www.w3.org/2000/svg",width:"14",height:"14",viewBox:"0 0 24 24",fill:"none",stroke:"currentColor",strokeWidth:"2",strokeLinecap:"round",strokeLinejoin:"round",children:[jsxRuntime.jsx("circle",{cx:"11",cy:"11",r:"8"}),jsxRuntime.jsx("path",{d:"m21 21-4.3-4.3"})]})}),jsxRuntime.jsx("input",{type:"text",placeholder:"Search records...",value:e.getState().globalFilter||"",onChange:l=>e.setGlobalFilter(l.target.value),className:"sdg__search-input"})]})}),jsxRuntime.jsxs("div",{className:"sdg__toolbar-group",children:[a&&jsxRuntime.jsx("button",{type:"button",onClick:()=>e.setColumnFilters([]),className:"sdg__btn",children:"Reset Filters"}),jsxRuntime.jsx("button",{type:"button",onClick:()=>n(!i),className:"sdg__btn",children:"Columns"}),s&&jsxRuntime.jsx("button",{type:"button",className:"sdg__btn sdg__btn--primary",onClick:()=>o?.("csv"),children:"Export"})]}),i&&jsxRuntime.jsx(ce,{table:e})]})};var pe=({customLoadingState:e})=>jsxRuntime.jsx("div",{className:"sdg__status",children:e||jsxRuntime.jsxs("div",{className:"sdg__toolbar-group",style:{flexDirection:"column"},children:[jsxRuntime.jsx("div",{className:"sdg__spinner"}),jsxRuntime.jsx("p",{className:"sdg__status-text",children:"Loading data..."})]})});var ue=({customEmptyState:e})=>jsxRuntime.jsx("div",{className:"sdg__status",children:e||jsxRuntime.jsx("div",{className:"sdg__status-text",children:jsxRuntime.jsx("p",{children:"No records found"})})});var _e=({error:e,customErrorState:t})=>jsxRuntime.jsx("div",{className:"sdg__status",children:t||jsxRuntime.jsxs("div",{className:"sdg__status-error",children:[jsxRuntime.jsx("p",{style:{fontWeight:700},children:"Error occurred"}),jsxRuntime.jsx("p",{style:{fontSize:"0.875rem"},children:typeof e=="string"?e:e.message})]})});var be=({table:e,actions:t,onBulkAction:a})=>{let s=e.getSelectedRowModel().rows.map(o=>o.original);return s.length===0?null:jsxRuntime.jsxs("div",{className:"sdg__bulk",children:[jsxRuntime.jsxs("div",{className:"sdg__bulk-meta",children:[jsxRuntime.jsx("span",{className:"sdg__bulk-count",children:s.length}),jsxRuntime.jsx("span",{children:"rows selected"})]}),jsxRuntime.jsxs("div",{className:"sdg__toolbar-group",children:[t.map((o,i)=>jsxRuntime.jsx("button",{type:"button",onClick:()=>a(o,s),className:`sdg__btn ${o.danger?"sdg__btn--danger":""}`,children:o.label},i)),jsxRuntime.jsx("button",{type:"button",onClick:()=>e.toggleAllRowsSelected(false),className:"sdg__btn sdg__btn--ghost",children:"Deselect All"})]})]})};var fe=({rowCount:e=10,colCount:t,density:a})=>{let s={compact:"sdg__skeleton-row--compact",comfortable:"sdg__skeleton-row--comfortable",spacious:"sdg__skeleton-row--spacious"}[a]||"sdg__skeleton-row--comfortable";return jsxRuntime.jsx("div",{className:"sdg__skeleton",children:Array.from({length:e}).map((o,i)=>jsxRuntime.jsx("div",{className:`sdg__skeleton-row ${s}`,children:Array.from({length:t}).map((n,l)=>jsxRuntime.jsx("div",{className:"sdg__skeleton-cell"},l))},i))})};function nt({data:e,columns:t,loading:a,error:s,emptyState:o,loadingState:i,errorState:n,searchable:l,filterable:G,pagination:b,selectable:p,exportable:R,toolbar:N,stickyHeader:f,virtualized:v,manualPagination:y,manualSorting:u,manualFiltering:_,rowActions:A,bulkActions:z,onBulkAction:B,onRowClick:I,onRowDoubleClick:c,onSelectionChange:he,onSortChange:ye,onFilterChange:j,onPaginationChange:xe,onExport:we,className:Se}){let ke=w(),{state:m,setSorting:Ce,setColumnFilters:Re,setGlobalFilter:Ne,setColumnVisibility:De,setRowSelection:Te,setPagination:Ge,density:Ee}=ke,[E,K]=Oe.useState(false),[Fe,Pe]=Oe.useState(null),Ae=Oe.useCallback(r=>{Pe(r);},[]);Oe.useEffect(()=>{let r=d=>{d.key==="Escape"&&E&&K(false);};return window.addEventListener("keydown",r),()=>window.removeEventListener("keydown",r)},[E]);let ee=Oe.useMemo(()=>p?[{id:"selection",header:({table:r})=>jsxRuntime.jsx("input",{type:"checkbox",className:"sdg__checkbox",checked:r.getIsAllPageRowsSelected(),onChange:r.getToggleAllPageRowsSelectedHandler(),ref:d=>{d&&(d.indeterminate=r.getIsSomePageRowsSelected());},"aria-label":"Select all rows"}),cell:({row:r})=>jsxRuntime.jsx("input",{type:"checkbox",className:"sdg__checkbox",checked:r.getIsSelected(),onChange:r.getToggleSelectedHandler(),"aria-label":"Select row"}),size:40},...t]:t,[t,p]),D=reactTable.useReactTable({data:e,columns:ee,state:{sorting:m.sorting,columnFilters:m.columnFilters,globalFilter:m.globalFilter,columnVisibility:m.columnVisibility,rowSelection:m.rowSelection,pagination:m.pagination},onSortingChange:r=>{let d=typeof r=="function"?r(m.sorting):r;Ce(d),ye?.(d);},onColumnFiltersChange:r=>{let d=typeof r=="function"?r(m.columnFilters):r;Re(d),j?.(d,m.globalFilter);},onGlobalFilterChange:r=>{let d=typeof r=="function"?r(m.globalFilter):r;Ne(d),j?.(m.columnFilters,d);},onColumnVisibilityChange:r=>{let d=typeof r=="function"?r(m.columnVisibility):r;De(d);},onRowSelectionChange:r=>{let d=typeof r=="function"?r(m.rowSelection):r;Te(d),he?.(d);},onPaginationChange:r=>{let d=typeof r=="function"?r(m.pagination):r;Ge(d),xe?.(d);},getCoreRowModel:reactTable.getCoreRowModel(),getPaginationRowModel:v?void 0:reactTable.getPaginationRowModel(),getSortedRowModel:reactTable.getSortedRowModel(),getFilteredRowModel:reactTable.getFilteredRowModel(),manualPagination:v?true:y,manualSorting:u,manualFiltering:_,enableRowSelection:p});return s?jsxRuntime.jsx(_e,{error:s,customErrorState:n}):a?jsxRuntime.jsx(pe,{customLoadingState:i||jsxRuntime.jsx(fe,{colCount:ee.length,density:Ee})}):e.length===0?jsxRuntime.jsx(ue,{customEmptyState:o}):jsxRuntime.jsxs("div",{className:`sdg ${E?"sdg--fullscreen":""} ${Se||""}`,children:[jsxRuntime.jsxs("div",{className:"sdg__chrome",children:[N&&jsxRuntime.jsx(me,{table:D,searchable:l,filterable:G,exportable:R,onExport:we}),jsxRuntime.jsx("button",{type:"button",onClick:()=>K(!E),className:"sdg__icon-btn",title:"Toggle Fullscreen",children:jsxRuntime.jsx("svg",{xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 24 24",fill:"none",stroke:"currentColor",strokeWidth:"2",strokeLinecap:"round",strokeLinejoin:"round",children:E?jsxRuntime.jsx("path",{d:"M15 3h6v6M9 21H3v-6M21 3l-7 7M3 21l7-7"}):jsxRuntime.jsx("path",{d:"M8 3H5a2 2 0 0 0-2 2v3m18 0V5a2 2 0 0 0-2-2h-3m0 18h3a2 2 0 0 0 2-2v-3M3 16V19a2 2 0 0 0 2 2h3"})})})]}),p&&jsxRuntime.jsx(be,{table:D,actions:z||[],onBulkAction:B||(()=>{})}),jsxRuntime.jsx("div",{ref:Ae,className:`sdg__scroll ${v?"sdg__scroll--virtual":""} ${f?"sdg__scroll--sticky-header":""}`,children:jsxRuntime.jsxs("table",{className:"sdg__table",children:[jsxRuntime.jsx(ae,{table:D}),v?jsxRuntime.jsx(de,{table:D,scrollElement:Fe,onRowClick:I,onRowDoubleClick:c,rowActions:A}):jsxRuntime.jsx(ne,{table:D,onRowClick:I,onRowDoubleClick:c,rowActions:A})]})}),b&&jsxRuntime.jsx(le,{table:D})]})}function wo({...e}){return jsxRuntime.jsx(se,{persistenceKey:e.persistenceKey,children:jsxRuntime.jsx(nt,{...e})})}
exports.DataGrid=wo;
